tests/test_transactions_api.py

```diff
@@ -626,3 +630,42 @@ def test_span_id(self) -> None:
         assert response.status_code == 200, response.data
 
         assert data["data"][0]["span_id"] == "8841662216cc598b"
+
+    def test_limitby_multicolumn(self) -> None:
+        query_str = """MATCH (transactions)
+                    SELECT project_id,
+                           environment,
+                           platform,
+                           event_id
+                    WHERE project_id = 1
+                    AND finish_ts >= toDateTime('{start_time}')
+                    AND finish_ts < toDateTime('{end_time}')
+                    LIMIT {limit_by_count} BY environment, platform
+                    """.format(
+            start_time=(self.base_time - self.skew).isoformat(),
+            end_time=(self.base_time + self.skew).isoformat(),
+            limit_by_count=LIMIT_BY_COUNT,
+        )
+        response = self.app.post(SNQL_ROUTE, data=json.dumps({"query": query_str}))
+        parsed_data = json.loads(response.data)
+
+        assert response.status_code == 200
+        # Note that the following assertions will be wrong if there isn't a sufficient cross-product
+        # of generated test data (meaning each environment needs to have at least
+        # LIMIT_BY_COUNT records present for each platform). The number of environments
+        # had to be made unequal to the number of platforms to get the test data generator
+        # to write suitable input data
```

Comment:
Let me know if we think this is too brittle. I wanted to avoid mucking around with the definition of `generate_fizzbuzz_events`

Reply:
This mostly makes sense, but I'm not sure I understand the last part. Can you elaborate on why adding another environment was necessary?

Reply:
If you look at the data generator, both platform and environment are set from an input array that's cycled through via the value of the variable `tock`:

```
"platform": self.platforms[
    (tock * p) % len(self.platforms)
],
```

```
"environment": self.environments[
    (tock * p) % len(self.environments)
]
```

If both of these are the same length, then every environment gets matched to one platform in every record (e.g. "a" is always "prod" and "b" is always "test"). This means that if you try to `LIMIT BY platform, environment` you can't test that individual column values are being used in the LIMIT BY (it's equivalent to `LIMIT BY platform` or `LIMIT BY environment`).

Does that make sense?

Reply:
Ah I see, yes that makes sense. Thank you.